src/devices/Ina219/Ina219.cs

```diff
@@ -0,0 +1,375 @@
+        private AdcResolutionOrSamples _shuntAdcResSamp;
+
+        /// <summary>
+        /// The CurrentLsb property specifies the electrical current in amperes respresented by the least significant bit of the INA219 Current register.
+        /// It is used by the GetCurrent method to calculate the current from the value exposed by the current register
+        /// <see cref="GetCurrent"/>
+        /// </summary>
+        /// <remarks>
+        /// This property defaults to 1. It is set by calling SetCalibration with values for the ADC sensitivity, current range and optionally shunt resistance.
+        /// If the value of the calibration register is set manually then this property will also need to be set for GetCurrent to return the correct value.
+        /// <see cref="SetCalibration(PgaSensitivity, float, float)"/> <seealso cref="SetCalibration(ushort)"/>
+        /// </remarks>
+        public float CurrentLsb { get; set; }
+
+        /// <summary>
+        /// The PowerLsb property specifies the electrical power in watts respresented by the least significant bit of the INA219 Power register.
+        /// It is used by the GetPower method to calculate the power from the value exposed by the Power register
+        /// <see cref="GetCurrent"/>
+        /// </summary>
+        /// <remarks>
+        /// This property defaults to 1. It is set by calling SetCalibration with values for the ADC sensitivity, current range and optionally shunt resistance.
+        /// If the value of the calibration register is set manually then this property will also need to be set for GetPower to return the correct value.
+        /// <see cref="SetCalibration(PgaSensitivity, float, float)"/> <seealso cref="SetCalibration(ushort)"/>
+        /// </remarks>
+        public float PowerLsb { get; set; }
+
+        /// <summary>
+        /// Construct an Ina219 device using an I2cDevice
+        /// </summary>
+        /// <remarks>
+        /// This binding does not dispose the passed in I2cDevice.
+        /// </remarks>
+        /// <param name="i2cDevice">The I2cDevice initialized to communicate with the Ina219.</param>
+        public Ina219(I2cDevice i2cDevice)
+        {
+            if (i2cDevice == null)
+            {
+                throw new System.ArgumentNullException(nameof(i2cDevice));
+            }
+
+            PowerLsb = CurrentLsb = 1F;
+
+            _i2cDevice = i2cDevice;
+        }
+
+        /// <summary>
+        /// Construct an Ina219 device using an I2cConnectionSettings.
+        /// </summary>
+        /// <remarks>
+        /// This binding creates an I2cDevice ufor communication with the Ina219. The I2cDevice is disposed when then Ina219 is disposed.
+        /// </remarks>
+        /// <param name="settings">The I2cConnectionSettings object initialized with the appropiate settings to communicate with the Ina219.</param>
+        public Ina219(I2cConnectionSettings settings)
```

Comment:
we usually don't add such overloads in other bindings but if you're adding one at least please implement it similarly to following:
```csharp
public Ina219(I2cConnectionSettings settings) : this(I2cDevice.Create(settings)
{
}
```

Reply:
I like the idea of a constructor  where you want the binding to manage the lifetime of the i2cdevice and one where it’s managed internally, but I take your point with the implementation.

Reply:
@krwq 

I had a look at this but because I wanted to add error checking to ensure that the passed in settings were non-null then the device would have been created before the check was performed.

I have added a private pseudo constructor to ensure that code isn't duplicated